#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

try:
    import imp
    imp.find_module(PACKAGE)
    from dynamic_reconfigure.parameter_generator_catkin import *;
except:
    import roslib; roslib.load_manifest(PACKAGE)
    from dynamic_reconfigure.parameter_generator import *;

from math import pi

gen = ParameterGenerator ()

algorithm_enum = gen.enum([gen.const("RANSAC", str_t, "RANSAC", "RANSAC"),
                           gen.const("LMEDS", str_t, "LMEDS", "LMEDS"),
                           gen.const("MSAC", str_t, "MSAC", "MSAC"),
                           gen.const("RRANSAC", str_t, "RRANSAC", "RRANSAC"),
                           gen.const("RMSAC", str_t, "RMSAC", "RMSAC"),
                           gen.const("MLESAC", str_t, "MLESAC", "MLESAC"),
                           gen.const("PROSAC", str_t, "PROSAC", "PROSAC"),
                           ],
                           "Available Algorithmes")
gen.add("algorithm", str_t, 0, "algorithm to estimate model", "RANSAC", edit_method=algorithm_enum)
gen.add("min_radius", double_t, 0, "minium radius of torus", 0.1, 0.0, 5.0)
gen.add("max_radius", double_t, 0, "minium radius of torus", 1.0, 0.0, 5.0)
gen.add("outlier_threshold", double_t, 0, "outlier threshold", 0.01, 0.0, 1.0)
gen.add("max_iterations", int_t, 0, "max iterations", 100, 0, 10000)
gen.add("min_size", int_t, 0, "minimum number of inliers", 10, 0, 1000)
gen.add("eps_hint_angle", double_t, 0, "eps angle for hint", 0.1, 0.0, pi)
gen.add("voxel_grid_sampling", bool_t, 0, "use voxel grid sampling", False)
gen.add("voxel_size", double_t, 0, "voxel grid size", 0.02, 0.001, 1.0)

exit (gen.generate (PACKAGE, "jsk_pcl_ros", "TorusFinder"))
